packages/dds/merge-tree/src/mergeTreeNodes.ts

```diff
@@ -651,18 +539,24 @@ export abstract class BaseSegment implements ISegment {
 		return true;
 	}
 
-	protected cloneInto(b: ISegment): void {
-		b.clientId = this.clientId;
+	protected cloneInto(seg: ISegment): void {
+		const b: ISegmentLeaf = seg;
+		if (isInserted(this)) {
+			b.clientId = this.clientId;
+			b.seq = this.seq;
+		}
 		// TODO: deep clone properties
 		b.properties = clone(this.properties);
-		b.removedClientIds = this.removedClientIds?.slice();
-		// TODO: copy removed client overlap and branch removal info
-		b.removedSeq = this.removedSeq;
-		b.movedClientIds = this.movedClientIds?.slice();
-		b.movedSeq = this.movedSeq;
-		b.movedSeqs = this.movedSeqs;
```

Comment:
this was a bug. we were not copying this array, not sure if there are any implications as we don't use clone much